File: 19th/코딩테스트 기초반/week6/10845번_큐.cpp
#include <iostream>
#include <queue>
#include <string>

using namespace std;

int main() {
    int N;
    queue<int> q;
    string cmd;

    cin >> N; // 명령의 수

    while (N--) {
        cin >> cmd; // 명령어

        if (cmd == "push") { // push 명령어
            int X;
            cin >> X;
            q.push(X);
        }

        if (cmd == "pop") { // pop 명령어
            if (q.empty()) cout << -1 << "\n";
            else {
                cout << q.front() << "\n";
                q.pop();
            }
        }

        if (cmd == "size") cout << q.size() << "\n"; // size 명령어

        if (cmd == "empty") { // empty 명령어
            if (q.empty()) cout << 1 << "\n";
            else cout << 0 << "\n";
        }

        if (cmd == "front") { // front 명령어
            if (q.empty()) cout << -1 << "\n";
            else cout << q.front() << "\n";
        }

        if (cmd == "back") { // back 명령어
            if (q.empty()) cout << -1 << "\n";
            else cout << q.back() << "\n";
        }
    }
}

File: 19th/코딩테스트 기초반/week6/10845번_큐.py
from collections import deque

q = deque()
for _ in range(int(input())):
    cmd = input()
    if cmd == 'pop':  # 큐의 맨 앞 원소 빼기
        if q:
            print(q.popleft())
        else:
            print(-1)
    elif cmd == 'size':  # 큐 크기 출력
        print(len(q))
    elif cmd == 'empty':  # 큐가 비었는지 확인
        if q:
            print(0)
        else:
            print(1)
    elif cmd == 'front':  # 큐의 맨 앞 원소 확인
        if q:
            print(q[0])
        else:
            print(-1)
    elif cmd == 'back':  # 큐의 맨 뒤 원소 확인
        if q:
            print(q[-1])
        else:
            print(-1)
    else:  # push X 명령 처리
        push, x = cmd.split()
        q.append(x)